src/App.tsx

```diff
@@ -15,47 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((goods1, goods2) => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return goods1.localeCompare(goods2);
+
+      case SortType.LENGTH:
+        return goods1.length - goods2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed ? visibleGoods.reverse() : visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
```

Comment:
Add generic type if you use custom or non-primitive data types.
```suggestion
  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
```